x-pack/plugins/security_solution/server/lib/detection_engine/rules/update_rules.ts

```diff
@@ -4,179 +4,94 @@
+      timelineId: ruleUpdate.timeline_id,
+      timelineTitle: ruleUpdate.timeline_title,
+      meta: ruleUpdate.meta,
+      maxSignals: ruleUpdate.max_signals ?? DEFAULT_MAX_SIGNALS,
+      riskScore: ruleUpdate.risk_score,
+      riskScoreMapping: ruleUpdate.risk_score_mapping ?? [],
+      ruleNameOverride: ruleUpdate.rule_name_override,
+      severity: ruleUpdate.severity,
+      severityMapping: ruleUpdate.severity_mapping ?? [],
+      threat: ruleUpdate.threat ?? [],
+      timestampOverride: ruleUpdate.timestamp_override,
+      to: ruleUpdate.to ?? 'now',
+      references: ruleUpdate.references ?? [],
+      note: ruleUpdate.note,
+      // Always use the version from the request if specified. If it isn't specified, leave immutable rules alone and
+      // increment the version of mutable rules by 1.
+      version:
+        ruleUpdate.version ?? existingRule.params.immutable
+          ? existingRule.params.version
+          : existingRule.params.version + 1,
+      exceptionsList: ruleUpdate.exceptions_list ?? [],
+      ...typeSpecificParams,
+    },
+    schedule: { interval: ruleUpdate.interval ?? '5m' },
+    actions: throttle === 'rule' ? (ruleUpdate.actions ?? []).map(transformRuleToAlertAction) : [],
+    throttle: null,
+  };
 
   const update = await alertsClient.update({
-    id: rule.id,
-    data: {
-      tags: addTags(tags, rule.params.ruleId, rule.params.immutable),
-      name,
-      schedule: { interval },
-      actions: actions.map(transformRuleToAlertAction),
-      throttle: null,
-      params: {
-        author,
-        buildingBlockType,
-        description,
-        ruleId: rule.params.ruleId,
-        falsePositives,
-        from,
-        immutable: rule.params.immutable,
-        query,
-        language,
-        license,
-        outputIndex,
-        savedId,
-        timelineId,
-        timelineTitle,
-        meta,
-        filters,
-        index,
-        maxSignals,
-        riskScore,
-        riskScoreMapping,
-        ruleNameOverride,
-        severity,
-        severityMapping,
-        threat,
-        threshold,
-        threatFilters,
-        threatIndex,
-        threatQuery,
-        threatMapping,
-        threatLanguage,
-        timestampOverride,
-        to,
-        type,
-        references,
-        note,
-        version: calculatedVersion,
-        anomalyThreshold,
-        machineLearningJobId,
-        exceptionsList,
-      },
-    },
+    id: existingRule.id,
+    data: newInternalRule,
   });
 
-  if (rule.enabled && enabled === false) {
-    await alertsClient.disable({ id: rule.id });
-  } else if (!rule.enabled && enabled === true) {
-    await alertsClient.enable({ id: rule.id });
+  if (existingRule.enabled && enabled === false) {
+    await alertsClient.disable({ id: existingRule.id });
+  } else if (!existingRule.enabled && enabled === true) {
+    await alertsClient.enable({ id: existingRule.id });
```

Comment:
Feel free if you want to directly use the booleans:

```ts
  if (existingRule.enabled && !enabled) {
    await alertsClient.disable({ id: existingRule.id });
  } else if (!existingRule.enabled && enabled) {
    await alertsClient.enable({ id: existingRule.id });
```

Instead of the explicit `=== true` or `=== false`, those are probably just older bad habits from too much time with JS types that _could_ be undefined or null, but I think these booleans are ok to use normally if you think that reads better.

No changes requested, just pointing out parts of code myself or others wrote that might read odd.

Reply:
If it helps, the alerting framework team helped us with this part as we have to put the enabling/disabling of rules _after_ we do any updates first because once we enable/disable a rule it will invalidate API keys and can cause a "false run" to happen where the first run has a bad API key.

So right in this section when we call this line:

```ts
await alertsClient.enable({ id: existingRule.id });
```

That is going to start the rule and invalidate any previous API keys it might have had. So just a FYI for you if you haven't figured that part out yet looking through here.